portfolio/blocks.py

```diff
@@ -0,0 +1,37 @@
+from wagtail.blocks import (
+    CharBlock,
+    ListBlock,
+    PageChooserBlock,
+    RichTextBlock,
+    StructBlock,
+)
+from wagtail.images.blocks import ImageChooserBlock
+
+from base.blocks import BaseStreamBlock
+
+
+class CardBlock(StructBlock):
+    heading = CharBlock()
+    # To improve consistency in how blocks appear, we only allow certain rich text features to be used.
+    # For example, headings aren’t available because the section will already have a heading.
+    text = RichTextBlock(features=["bold", "italic", "link"])
+    image = ImageChooserBlock(required=False)
+
+    class Meta:
+        icon = "form"
+        template = "portfolio/blocks/card_block.html"
+
+
+class FeaturedPostsBlock(StructBlock):
+    heading = CharBlock()
+    text = RichTextBlock(features=["bold", "italic", "link"], required=False)
+    posts = ListBlock(PageChooserBlock(page_type="blog.BlogPage"))
```

Comment:
Blog's are a great way to demonstrate the `RoutablePageMixin`with custom paths. 

/blog/all/ 
/blog/tag/{tag-name}/
/blog/year/{year}/ 

Simple examples could be helpful. Emphasis on _simple_.

Reply:
👍 that’d be great! And aligns _ok_ with our goal of matching the bakerydemo (https://github.com/wagtail/bakerydemo/blob/2edb8745895916f6090662f13f7e35600fb36727/bakerydemo/blog/models.py#L180)